Firmware/src/libs/RingBuffer.h

```diff
@@ -0,0 +1,266 @@
+//
```

Comment:
This seems to be a direct copy of the planner queue, not really an efficient use. Does not need the movement through the queues either which is not standard for a ringbuffer. Should ONLY have push and pop to be truly thread safe.